File: 7-kyu/a-rule-of-divisibility-by-7/python/solution.py
from math import log10

def seven(m):
    steps = 0
    if m < 1:
        return (m, steps)
    while log10(m) > 2:
        m = (m // 10) - ((m % 10) * 2)
        steps += 1
        if m < 1:
            break
    return (m, steps)

File: 7-kyu/a-rule-of-divisibility-by-7/javascript/solution.js
const seven = m => {
  let steps = 0;
  if (m < 1) {
    return [m, steps];
  }
  while (m >= 100) {
    m = Math.floor(m / 10) - (m % 10) * 2;
    steps++;
  }
  return [m, steps];
}